pinot/net.py

```diff
@@ -56,28 +112,21 @@ def condition(self, g):
         # get the parameters
         theta = self.forward(g)
 
+        # put theta through noise model
+        theta = self.noise_model(theta)
+
         # parameterize the distribution according to the parameters
         # and the distribution class
-
-        distribution = self.distribution_class(
-            *self.param_transform(*torch.unbind(theta, dim=-1))
-        )
+        distribution = self.distribution_class(*theta)
```

Comment:
why not use the noise model flag here and drop this contrived distribution_class object?
It would simplify so many things above...

Is there a strong reason to keep this?

Reply:
well it's not necessarily Normal

Reply:
the noise model 'homoschedastic_normal' implies normal with fixed global variance.
The distribution and its parametrization are both properties of noise_model.
If you want another distribution, you are picking another likelihood which is another noise model. 

I would put all this into noise model or likelihood model or however you want to call it.

Passing the distribution separately makes no sense, these two objects are bound together.

Reply:
I would absorb everything into noise model like this, as I also suggested a long time ago in the modeling issue:
```
def forward/condition/whatever():
    if noise_model=='normal_something_schedastic'
        mu = ...
        log_sigma = ....
        sigma = torch.exp(sigma)
        distribution = Normal(mu,sigma)
    if noise_model == 'categorical':
        alpha = ... #i.e. output from a regressor
        dist_noise = Categorical(alpha)
```

Reply:
A HUGE advantage of this is that we keep the likelihood very interpretable code-wise ad do not mix it with representation stuff.

I think we will have another small refactoring step here later on when we try things that are not simple linear decoders/regressors, but we can cross that bridge when we get there.

Reply:
@yuanqing-wang  so I have this alternative pitch if you prefer it.

You can create a meta-object which ties together parametrization and distribution and passes them in here instead of a flag if you prefer.

i.e. outside of the model you can create that thing as
```
class HomoSchedasticNormal():
    def __init()__:
    .....properties and parametrizations
 ```